proto/accessgraph/v1alpha/access_graph_service.proto

```diff
@@ -610,6 +632,96 @@ message AWSCloudTrailEventResource {
   string type = 2;
 }
 
+// KubeAuditLogStreamRequest is sent from the client (Teleport) to the server
+// (access graph) over the bi-directional KubeAuditLogStream. It encapsulates
+// distinct client actions for sending batches of Kubernetes audit log events
+// and managing the persistent resume state maintained by the access graph
+// server.
+message KubeAuditLogStreamRequest {
+  oneof action {
+    KubeAuditLogConfig config = 1;
+    KubeAuditLogEvents events = 2;
+  }
+}
+
+// KubeAuditLogConfig is the KubeAuditLogStream configuration. It is sent from
+// the client (Teleport) to the server (access graph) on start up, containing
+// any configuration parameters from the Teleport config. The server will
+// respond with an adjusted config that is that it will use, possibly
+// overriding the supplied config.
+message KubeAuditLogConfig {}
+
+// KubeAuditLogEvents is a chunk of Kubernetes apiserver audit logs.
+message KubeAuditLogEvents {
+  // event is a Kubernetes audit log event represented as a struct so as to
+  // not depend on the Kubernetes proto files. Additionally, the audit log
+  // events are typically received as JSON, not protobufs.
+  // package k8s.io.apiserver.pkg.apis.audit.v1 at
+  // https://github.com/kubernetes/apiserver/blob/master/pkg/apis/audit/v1/generated.proto
+  repeated google.protobuf.Struct events = 1;
+
+  // cursor is the position in the event stream of the last event in the
+  // event field. It can be sent back to Teleport when a stream is
+  // reconnected to resume the event stream from the end of this chunk.
+  KubeAuditLogCursor cursor = 2;
+}
+
+// KubeAuditLogCursor is a position in a stream of KubeAuditLogEvents for
+// resuming a stream of events after an interrupted connection. Each batch
+// of events has a cursor referring to the last event of the batch. The
+// cursor can be sent to Teleport when resuming a connection.
+message KubeAuditLogCursor {
+  // KubeAuditLogSource specifies a source of logs in order to interpret a
+  // KubeAuditLogCursor id and cursor. Different sources of logs will have
+  // different ways of representing a position in a log stream.
+  enum KubeAuditLogSource {
+    // KUBE_AUDIT_LOG_SOURCE_UNSPECIFIED represents a zero value where no log
+    // source has been specified.
+    KUBE_AUDIT_LOG_SOURCE_UNSPECIFIED = 0;
+    // KUBE_AUDIT_LOG_SOURCE_AWS_CLOUDWATCH specifies AWS CloudWatch as the
+    // source of Kubernetes apiserver audit logs. This is where EKS sends
+    // the apiserver audit logs.
+    KUBE_AUDIT_LOG_SOURCE_AWS_CLOUDWATCH = 1;
```

Comment:
Perhaps this should be KUBE_AUDIT_LOG_SOURCE_EKS_AWS_CLOUDWATCH - i.e. add "EKS" in there. Or perhaps just EKS as cloudwatch is the way EKS delivers their audit logs.

It is possible we would bypass cloudwatch by having our own agent running in the cluster on EKS as reports are the cloudwatch logs get expensive due to the volume of logs. That would be a different log source but still be EKS.

Reply:
Ok, I'll go with the shorter name. it's not really the most accurate way to name it, but I think it is sufficient.